src/box/lua/iproto.c

```diff
@@ -375,6 +378,71 @@ lbox_iproto_send(struct lua_State *L)
 	return (rc == 0) ? 0 : luaT_error(L);
 }
 
+/**
+ * Lua request handler callback: creates new Lua execution context, gets the Lua
+ * callback function, pushes the request header and body as MsgPack objects and
+ * calls the Lua callback.
+ */
+static enum iproto_handler_status
+lua_rq_handler_cb(const char *header, const char *header_end,
+		  const char *body, const char *body_end,
+		  void *ctx)
+{
+	struct lua_State *L = luaT_newthread(tarantool_L);
+	if (L == NULL)
+		return IPROTO_HANDLER_ERROR;
+	int coro_ref = luaL_ref(tarantool_L, LUA_REGISTRYINDEX);
+	int cb_ref = (int)(uintptr_t)ctx;
+	lua_rawgeti(L, LUA_REGISTRYINDEX, cb_ref);
+	luamp_push_with_dict(L, header, header_end, iproto_key_translation);
+	luamp_push_with_dict(L, body, body_end, iproto_key_translation);
+	if (luaT_call(L, 2, 1) != 0)
+		return IPROTO_HANDLER_ERROR;
+	bool ok = lua_toboolean(L, 1);
+	luaL_unref(tarantool_L, LUA_REGISTRYINDEX, coro_ref);
+	return ok ? IPROTO_HANDLER_OK : IPROTO_HANDLER_FALLBACK;
+}
+
+/**
+ * Lua request handler destructor: unreferences the request handler's Lua
+ * callback function.
+ */
+static void
+lua_rq_handler_dtor(void *ctx)
+{
+	luaL_unref(tarantool_L, LUA_REGISTRYINDEX, (int)(uintptr_t)ctx);
+}
+
+/**
+ * Sets IPROTO request handler callback (second argument) for the given request
+ * type (first argument): the Lua callback function is referenced in Lua and
+ * unreferenced in `lua_rq_handler_dtor`.
+ * Passing nil as the callback resets the corresponding request handler.
+ */
+static int
+lbox_iproto_override(struct lua_State *L)
+{
+	int n_args = lua_gettop(L);
+	if (n_args != 2)
```

Comment:
Why not allow to omit the second argument? It's common in Lua. You could use `lua_isnoneornil` instead of `lua_isnil` below.